### PR TITLE
Respect user specified platform and configuration #11

### PR DESCRIPTION
**Pull request by amrmahdi**

### CODE REVIEW

File: sln.cmd [hunk 1]
```diff
@@ -3,7 +3,7 @@ SETLOCAL
 
 SET MSBUILD_PROJECT=%~dp0src\SlnGen.Build.Tasks.UnitTests\SlnGen.Build.Tasks.UnitTests.csproj
 
-SET MSBUILD_ARGS=/NoLogo /Restore /Verbosity:Minimal /Target:SlnGen
+SET MSBUILD_ARGS=/NoLogo /Target:Restore /Verbosity:Minimal /Target:SlnGen
```
Comment (jeffkl) [inline sln.cmd]: Why did you make this change?

File: src/SlnGen.Build.Tasks/Internal/SlnFile.cs [hunk 2]
```diff
@@ -122,27 +110,33 @@ public void Save(TextWriter writer)
             writer.WriteLine("Global");
 
             writer.WriteLine("	GlobalSection(SolutionConfigurationPlatforms) = preSolution");
-            foreach (string configuration in _configurations)
+
+            IEnumerable<string> globalConfigurations = new HashSet<string>(_projects.SelectMany(p => p.Configurations)).Distinct();
```
Comment (jeffkl) [inline src/SlnGen.Build.Tasks/Internal/SlnFile.cs]: I know VS has logic to parse property groups like this:
```xml
<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
```

Do you think its worth parsing conditions like that to see what the project is configured against?

Comment (amrmahdi) [reply src/SlnGen.Build.Tasks/Internal/SlnFile.cs]: This is actually how it is originally parsed. This is read from ConditiondProperties. See the extension method GetConditionedPropertyValuesOrDefault

File: sln.cmd (hunk 1 above)
Comment (amrmahdi) [reply sln.cmd]: MSBUILD : error MSB1001: Unknown switch.
Switch: /Restore

For switch syntax, type "MSBuild /help"

File: src/SlnGen.Build.Tasks/Internal/SlnFile.cs (hunk 2 above)
Comment (amrmahdi) [reply src/SlnGen.Build.Tasks/Internal/SlnFile.cs]: https://github.com/Microsoft/msbuild/blob/03d1435c95e6a85fbf949f94958e743bc44c4186/src/Build/Definition/Project.cs#L706

**Pull request merged**